src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java

```diff
@@ -105,4 +107,32 @@ public void execute_validInput_success() {
         GenerateLocationsCommand command = new GenerateLocationsCommand(index);
         assertCommandSuccess(command, model, expectedMessage, expectedModelForGenerate);
     }
+
+    @Test
+    public void execute_validInputFromViewingAllInfected_success() {
+        String expectedMessage = "Generated locations for: Daniel Meier";
+        Model modelForAllInfected = model;
+        modelForAllInfected.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        Model expectedModelForGenerate = expectedModel;
+        Predicate<Location> locationPredicate = location -> location.getId().getOneBased() == 6;
+        expectedModelForGenerate.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        expectedModelForGenerate.updateFilteredLocationList(locationPredicate);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertCommandSuccess(command, modelForAllInfected, expectedMessage, expectedModelForGenerate);
+    }
+
+    @Test
+    public void execute_invalidInputFromViewingAllQuarantined_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_IS_NOT_INFECTED;
+        Model modelForAllQuarantined = model;
+        modelForAllQuarantined.updateFilteredPersonList(PREDICATE_SHOW_ALL_QUARANTINED);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        try {
+            command.execute(modelForAllQuarantined);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
```

Comment:
Suggest to replace with AssertThrow from command utils

Reply:
Done!